src/StackExchange.Redis.Extensions.Core/StackExchangeRedisCacheClient.cs

```diff
@@ -628,8 +628,13 @@ public IEnumerable<string> SearchKeys(string pattern)
 
 			foreach (var endpoint in endPoints)
 			{
-				var dbKeys = Database.Multiplexer.GetServer(endpoint).Keys(Database.Database, pattern);
+				var server = Database.Multiplexer.GetServer(endpoint);
+				if (!server.IsConnected || !server.Features.Scan)

```

Comment:
Why not? The server **doesn't** support this operation meaning our the code
is doing something wrong and not supported.
An exception should be thrown (as this is an exceptional and unexpected
behavior) to alert the developer that some thing he/she is wrong and
unexpected here.
On Mar 16, 2016 03:07, "Aishel M" <notifications@github.com> wrote:

> In
> src/StackExchange.Redis.Extensions.Core/StackExchangeRedisCacheClient.cs
> <https://github.com/imperugo/StackExchange.Redis.Extensions/pull/59#discussion_r56268221>
> :
>
> > @@ -628,8 +628,13 @@ public IEnumerable<string> SearchKeys(string pattern)
> >
> >  			foreach (var endpoint in endPoints)
> >  			{
> > -				var dbKeys = Database.Multiplexer.GetServer(endpoint).Keys(Database.Database, pattern);
> > +				var server = Database.Multiplexer.GetServer(endpoint);
> > +				if (!server.IsConnected || !server.Features.Scan)
>
> We check for *server.IsConnected*: this check is also made internally but
> since the idea here is to iterate through various servers, having one
> offline shouldn't disrupt the entire operation.
> We also check for *server.Features.Scan*: this check is important in the
> scenario where scanning of keys has been disallowed in one of the servers,
> in which case the entire operation shouldn't be disrupted either.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/imperugo/StackExchange.Redis.Extensions/pull/59/files/39e2a9f1b027973caf274d99a1769560ff41423d#r56268221>
>